context/app/api/vitessce_confs/assay_confs.py

```diff
@@ -296,14 +296,24 @@ def get_conf_cells(self):
 
 
 _assays = None
+tc = None
 
 
 def _get_assay(data_type):
     "Return the assay class for the given data type"
     global _assays
-    if _assays is None:
+    global tc
+    if _assays is None and tc is None:
         tc = TypeClient(current_app.config["TYPE_SERVICE_ENDPOINT"])
         _assays = {assay.name: assay for assay in tc.iterAssays()}
+    # iterAssays only returns current, and not deprecated, assays,
+    # which may be present in the entity.
+    # getAssayType can check for deprecated assays properly.
+    if data_type not in _assays:
+        assay = tc.getAssayType(data_type)
+        # The assay.name in this case would be the new name for the
+        # data_type so we cache the result using the data_type from the entity.
+        _assays[data_type] = assay
```

Comment:
```suggestion
    global _type_client
    if _type_client is None:
        _type_client = TypeClient(current_app.config["TYPE_SERVICE_ENDPOINT"])
    
    global _assays
    if _assays is None:
        # iterAssays does not include deprecated assay names...
        _assays = {assay.name: assay for assay in _type_client.iterAssays()}
        
    if data_type not in _assays:
        # ... but getAssayType does handle deprecated names:
        _assays[data_type] = tc.getAssayType(data_type)
```
This might be easier to read, but I don't think I've changed the logic.
- Why the typeclient global? Preference to hide details. I might be missing something.
- Split up the `_assays is None and tc is None`, since there are really two separate things happening inside.

Reply:
If `type_client` weren't global (as things stood previously), we couldn't have accessed it after the first `if` block.  I suppose we could instantiate it every time.  I agree with the last point.